### PR TITLE
Add benzene / naphthalene test system

### PR DESCRIPTION
**Pull request by jchodera**
This adds a simple test system in which benzene is converted into naphthalene in vacuum and explicit solvent.

TODO:
- [x] add harness to run this system with varying number of NCMC steps (0, 10, 100, 1000, 10000, 100000)
- [x] add analysis code to analyze data

### CODE REVIEW

**Comment by jchodera** [comment]
I'm running into `nan` during Langevin dynamics:
```
********************************************************************************
MultiTargetDesign sampler iteration        6
================================================================================
SAMS sampler iteration     6
--------------------------------------------------------------------------------
Expanded Ensemble sampler iteration        6
................................................................................
MCMC sampler iteration 6
Taking 500 steps of Langevin dynamics...
Using platform 'CUDA'
potential  = 21723.744 kT
kinetic    = 3085255.979 kT
force norm = 6748.740 kT/A/dof
Traceback (most recent call last):
  File "perses/tests/testsystems.py", line 2106, in <module>
    run_fused_rings()
  File "perses/tests/testsystems.py", line 2097, in run_fused_rings
    testsystem.designer.run(niterations=100)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1181, in run
    self.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1163, in update
    self.update_samplers()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1124, in update_samplers
    sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1032, in update
    self.update_sampler()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 996, in update_sampler
    self.sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 862, in update
    self.update_positions()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 695, in update_positions
    self.sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 480, in update
    integrator.step(self.nsteps)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/simtk/openmm/openmm.py", line 4731, in step
    return _openmm.LangevinIntegrator_step(self, steps)
Exception: Particle coordinate is nan
```
Trying GHMC instead...

**Comment by jchodera** [comment]
Hm. Something must be wrong with the NCMC GHMC, since this really shouldn't happen:
```
Performing NCMC insertion
Traceback (most recent call last):
  File "perses/tests/testsystems.py", line 2106, in <module>
    run_fused_rings()
  File "perses/tests/testsystems.py", line 2097, in run_fused_rings
    testsystem.designer.run(niterations=100)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1310, in run
    self.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1292, in update
    self.update_samplers()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1253, in update_samplers
    sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1161, in update
    self.update_sampler()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1125, in update_sampler
    self.sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 992, in update
    self.update_state()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 918, in update_state
    [ncmc_new_positions, ncmc_introduction_logp, potential_insert] = self.ncmc_engine.integrate(topology_proposal, geometry_new_positions, direction='insert', iteration=self.iteration)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 446, in integrate
    logP_alchemical_correction = self._computeAlchemicalCorrection(unmodified_system, alchemical_system, initial_positions, final_positions, direction=direction)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 210, in _computeAlchemicalCorrection
    alchemical_potential_correction = computePotentialEnergy(unmodified_system, final_positions) - computePotentialEnergy(alchemical_system, final_positions)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 197, in computePotentialEnergy
    context.setPositions(positions)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/simtk/openmm/openmm.py", line 13580, in setPositions
    return _openmm.Context_setPositions(self, positions)
Exception: Particle coordinate is nan
```

**Comment by jchodera** [comment]
Actually, the particle coordinates are not going `nan` *during* NCMC integration---that error happens at `context.setPositions()` in [this line](https://github.com/choderalab/perses/blob/ce7f9eb05391e17b8a0c7d5958dbc007bd01d8b3/perses/annihilation/ncmc_switching.py#L210).

**Comment by jchodera** [comment]
Do we add a `MonteCarloBarostat` anywhere? I'm wondering if NCMC might be encountering box volume changes, which the current system isn't designed to handle.

**Comment by jchodera** [comment]
There doesn't seem to be a barostat added anywhere, which is something we should fix.

cc: https://github.com/choderalab/perses/issues/225

**Comment by jchodera** [comment]
I've added some `assert` checks for `NaN`, and it seems like this is indeed happening *during* NCMC GHMC integration, which I don't understand:
```
Performing NCMC insertion
Traceback (most recent call last):
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 415, in integrate
    assert quantity_is_finite(positions) == True
AssertionError
Traceback (most recent call last):
  File "perses/tests/testsystems.py", line 2106, in <module>
    run_fused_rings()
  File "perses/tests/testsystems.py", line 2097, in run_fused_rings
    testsystem.designer.run(niterations=100)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1318, in run
    self.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1300, in update
    self.update_samplers()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1261, in update_samplers
    sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1169, in update
    self.update_sampler()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1133, in update_sampler
    self.sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1000, in update
    self.update_state()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 926, in update_state
    [ncmc_new_positions, ncmc_introduction_logp, potential_insert] = self.ncmc_engine.integrate(topology_proposal, geometry_new_positions, direction='insert', iteration=self.iteration)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 430, in integrate
    raise e
AssertionError
```

**Comment by jchodera** [comment]
After adding some more debug info, it seems that NCMC GHMC fails when the initial configurations are ridiculously high energy during insertion:
```
Performing NCMC insertion
NCMC step        0  /        1   insert : Eold   4.04216166e+09 Enew   4.04216166e+09 work   0.00000000e+00
Traceback (most recent call last):
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 417, in integrate
    assert quantity_is_finite(positions) == True
AssertionError
Traceback (most recent call last):
  File "perses/tests/testsystems.py", line 2106, in <module>
    run_fused_rings()
  File "perses/tests/testsystems.py", line 2097, in run_fused_rings
    testsystem.designer.run(niterations=100)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1318, in run
    self.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1300, in update
    self.update_samplers()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1261, in update_samplers
    sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1169, in update
    self.update_sampler()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1133, in update_sampler
    self.sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1000, in update
    self.update_state()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 926, in update_state
    [ncmc_new_positions, ncmc_introduction_logp, potential_insert] = self.ncmc_engine.integrate(topology_proposal, geometry_new_positions, direction='insert', iteration=self.iteration)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 432, in integrate
    raise e
AssertionError
```
This suggests we should monitor the final energies coming out of RJMC to see if this is due to internal strain or if it is instead due to some weird failure in the alchemical modifications to turn off nonbonded interactions properly.

**Comment by jchodera** [comment]
I found an issue with how the NCMC code was setting the various alchemical `lambda` components. Previously, it had just been setting them all to zero before insertion, which is not what we wanted since some bonds and angles were turned off then. Instead, the alchemical schedule encoded in the `functions` block should have been used to set the initial values:
```python
default_functions = {
    'lambda_sterics' : 'lambda',
    'lambda_electrostatics' : 'lambda',
    'lambda_bonds' : '0.9*lambda + 0.1', # don't fully soften bonds
    'lambda_angles' : '0.9*lambda + 0.1', # don't fully soften angles
    'lambda_torsions' : 'lambda'
    }
```
I have a temporary fix in that uses a block of numerical expression parsing code since we'd otherwise need to evaluate these expressions with Lepton (which I don't believe is exposed via the Python API). I'm re-running the examples now to see if another failure will occur.

The code is currently slowed down by a bunch of debugging checks, but we can remove these once we verify things are working as expected.

**Comment by jchodera** [comment]
I've run into a `nan` during an NCMC `delete`:
```
Performing NCMC annihilation
Initial potential of 'delete' operation: unmodified potential was 26748.536 kT, alchemical potential was 26749.005 kT before changing lambda, 26749.005 kT after changing lambda
NCMC step        0  /       10   delete : Eold  -2.05591429e+05 Enew  -2.05640424e+05 work  -1.96425037e+01
Traceback (most recent call last):
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 452, in integrate
    assert quantity_is_finite(positions) == True
AssertionError
Traceback (most recent call last):
  File "perses/tests/testsystems.py", line 2106, in <module>
    run_fused_rings()
  File "perses/tests/testsystems.py", line 2097, in run_fused_rings
    testsystem.designer.run(niterations=100)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1318, in run
    self.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1300, in update
    self.update_samplers()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1261, in update_samplers
    sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1169, in update
    self.update_sampler()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1133, in update_sampler
    self.sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1000, in update
    self.update_state()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 897, in update_state
    [ncmc_old_positions, ncmc_elimination_logp, potential_delete] = self.ncmc_engine.integrate(topology_proposal, positions, direction='delete', iteration=self.iteration)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 467, in integrate
    raise e
AssertionError
```
What is odd here is that the old and new energies (`Eold` and `Enew`) were reasonable during this step, but somehow the positions ended up as `nan` after that *same* step. The debug block looks like this:
```
                    # DEBUG
                    Eold = integrator.getGlobalVariableByName("Eold")
                    Enew = integrator.getGlobalVariableByName("Enew")
                    print('NCMC step %8d  / %8d %8s : Eold %16.8e Enew %16.8e work %16.8e' % (step, self.nsteps, direction, Eold, Enew, work[step+1]))
                    positions = context.getState(getPositions=True).getPositions(asNumpy=True)
                    assert quantity_is_finite(positions) == True
```
I'm not sure I understand how this could happen!

**Comment by jchodera** [comment]
OK, the positions are definitely becoming `nan` but the acceptance criteria doesn't pick up on this and the energies are still finite:
```
Initial potential of 'delete' operation: unmodified potential was 32176.330 kT, alchemical potential was 32176.850 kT before changing lambda, 32176.850 kT after changing lambda
NCMC step        0  /        1   delete : Eold   1.43357857e+06 Enew   1.42241834e+06 work  -4.47421611e+03 xsum_old   4.94437012e+03 xsum_new              nan
Traceback (most recent call last):
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 454, in integrate
    assert quantity_is_finite(positions) == True
AssertionError
Traceback (most recent call last):
  File "perses/tests/testsystems.py", line 2106, in <module>
    run_fused_rings()
  File "perses/tests/testsystems.py", line 2097, in run_fused_rings
    testsystem.designer.run(niterations=100)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1318, in run
    self.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1300, in update
    self.update_samplers()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1261, in update_samplers
    sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1169, in update
    self.update_sampler()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1133, in update_sampler
    self.sampler.update()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 1000, in update
    self.update_state()
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/samplers/samplers.py", line 897, in update_state
    [ncmc_old_positions, ncmc_elimination_logp, potential_delete] = self.ncmc_engine.integrate(topology_proposal, positions, direction='delete', iteration=self.iteration)
  File "/cbio/jclab/home/chodera/miniconda/lib/python2.7/site-packages/perses/annihilation/ncmc_switching.py", line 469, in integrate
    raise e
AssertionError
```

**Comment by jchodera** [comment]
This may actually be harder to debug because there are *two* GHMC steps per call the first time that `integrator.step(1)` is called. I'll try to disable that.

File: perses/annihilation/ncmc_switching.py [hunk 1]
```diff
@@ -135,6 +136,29 @@ def _getAvailableParameters(self, system, prefix='lambda'):
...
+            The Context
+        functions : dict
+            A dictionary of functions
+        value : float
+            The alchemical lambda value
+
+        TODO: Improve function evaluation to better match Lepton and be more flexible in exact replacement of 'lambda' tokens
+
+        """
+        from parsing import NumericStringParser
```
Comment (juliebehr) [inline perses/annihilation/ncmc_switching.py]:
```
ImportError: No module named parsing
```
What is this? I can't conda install it

Comment (jchodera) [inline perses/annihilation/ncmc_switching.py]: `pyparsing`: https://anaconda.org/anaconda/pyparsing

```
conda install --yes pyparsing
```
This is just a stop-gap measure. We'll need some way to evaluate the functions we feed to lepton to set the individual context parameters `lambda_*` based on `lambda` to 0 or 1 and the beginning and end of the protocol, unless I can think of a clever way to use a `CustomIntegrator` to do the calculation we need.

Comment (jchodera) [inline perses/annihilation/ncmc_switching.py]: Actually, @juliebehr, I think I fucked up the imports if you're using python 3. Sorry about that!  Fixed now, I hope.

**Comment by pgrinaway** [comment]
Taking a look now.

**Comment by pgrinaway** [comment]
Ok, so switching to `FFAllAngleGeometryEngine`, I haven't observed the `NaN` issue. However, there may be a bug in reporting of the work (if there is one, it might propagate to the calculation of the acceptance criterion). I see lines like this:

`insert : Eold  -1.28014642e+04 Enew  -1.28049103e+04 work  -8.01169874e+04`

for the first iteration. The accumulated work in that column is also displayed as being a part of the acceptance criterion:

`NCMC step     9999  /    10000   insert : Eold  -1.40145256e+04 Enew  -1.40145466e+04 work  -8.01205256e+04 xsum_old   4.91630078e+03 xsum_new   4.91633252e+03 xsum   3.88672133e-03`


`logp_accept = -3.0953e+05 [logp_proposal +0.0000e+00 geometry_logp -4.1078e+05 switch_logp -5.1950e+02 ncmc_elimination_logp +2.1653e+04 ncmc_introduction_logp +8.0119e+04 old_log_weight +0.0000e+00 new_log_weight +0.0000e+00]
`

File: devtools/conda-recipe/meta.yaml [hunk 2]
```diff
@@ -40,7 +40,7 @@ requirements:
     - openmm
     - parmed
     - openmoltools
-    - alchemy
+    - alchemy >=1.1.3
```
Comment (pgrinaway) [inline devtools/conda-recipe/meta.yaml]: Nothing wrong here, but does the current conda version of alchemy have the `softcore_beta` parameter? I had some issue with that, and had to install directly from the git repo. Is there no `alchemy-dev`? I couldn't get that either.

File: perses/annihilation/ncmc_switching.py [hunk 3]
```diff
@@ -636,6 +713,11 @@ def getStatistics(self, context):
         else:
             return (0,0)
```
Comment (pgrinaway) [inline perses/annihilation/ncmc_switching.py]: Why do we return this?

File: perses/samplers/samplers.py [hunk 4]
```diff
@@ -363,6 +371,133 @@ def has_nan(self):
             return False
 
 ################################################################################
+# GHMC INTEGRATOR (TEMPORARY UNTIL MOVED TO OPENMMTOOLS)
```
Comment (pgrinaway) [inline perses/samplers/samplers.py]: Isn't this [already](https://github.com/choderalab/openmmtools/blob/master/openmmtools/integrators.py#L527) in `openmoltools`? I guess that one needs to be fixed/have the `select` statement inserted?

Comment (pgrinaway) [inline perses/samplers/samplers.py]: Oh, this one does too.

File: perses/annihilation/ncmc_switching.py [hunk 5]
```diff
@@ -885,9 +977,11 @@ def __init__(self, temperature, system, functions, nsteps=0, steps_per_propagati
             self.addAlchemicalPerturbationStep()
             self.addComputeGlobal("Enew", "energy")
             self.addComputeGlobal("total_work", "total_work + (Enew-Eold)")
-            self.beginWhileBlock('pstep < psteps')
+            #self.beginWhileBlock('pstep < psteps')
+            self.addComputeSum("xsum_old", "x") # DEBUG
             self.addGHMCStep()
-            self.addComputeGlobal('pstep', 'pstep+1')
-            self.endBlock()
+            self.addComputeSum("xsum_new", "x") # DEBUG
```
Comment (pgrinaway) [inline perses/annihilation/ncmc_switching.py]: I think these `xsum` computations are unnecessary, right?

File: perses/annihilation/ncmc_switching.py [hunk 6]
```diff
@@ -135,6 +143,30 @@ def _getAvailableParameters(self, system, prefix='lambda'):
...
+        functions : dict
+            A dictionary of functions
+        value : float
+            The alchemical lambda value
+
+        TODO: Improve function evaluation to better match Lepton and be more flexible in exact replacement of 'lambda' tokens
+        Perhaps this could be done with a special integrator?
+
+        """
+        from perses.annihilation import NumericStringParser
```
Comment (pgrinaway) [inline perses/annihilation/ncmc_switching.py]: Can we just set the initial value at `step = 0` inside the integrator?

File: perses/annihilation/ncmc_switching.py [hunk 7]
```diff
@@ -340,16 +377,27 @@ def integrate(self, topology_proposal, initial_positions, direction='insert', pl
...
+            self._updateAlchemicalState(context, functions, 0)
         elif direction == 'delete':
-            for parameter_name in available_parameters:
-                context.setParameter(parameter_name, 1)
+            self._updateAlchemicalState(context, functions, 1)
 
         # Compute initial potential of alchemical state.
         initial_potential = self.beta * context.getState(getEnergy=True).getPotentialEnergy()
+        # DEBUG
+        if self.verbose:
```
Comment (pgrinaway) [inline perses/annihilation/ncmc_switching.py]: I don't know if you want to keep this section just in case.

**Comment by pgrinaway** [comment]
Ok, I've left a few comments. Otherwise it looks fine to me.

File: devtools/conda-recipe/meta.yaml (hunk 2 above)
Comment (jchodera) [inline devtools/conda-recipe/meta.yaml]: `softcore_beta` was added in 1.2.0, so I'll update this:
https://github.com/choderalab/alchemy/releases

File: perses/annihilation/ncmc_switching.py (hunk 6 above)
Comment (jchodera) [inline perses/annihilation/ncmc_switching.py]: We already do, but we need some way to measure the work (potential energy change) for going from fully interacting system -> alchemically-modified system with `lambda = 1` -> alchemically-modified system with `lambda` at its starting value. We can do this by adding some more variables inside of the integrator that we can then access after the fact, though. I'll see if I can add this and remove this extra code.

File: perses/annihilation/ncmc_switching.py (hunk 7 above)
Comment (jchodera) [inline perses/annihilation/ncmc_switching.py]: I've protected these inside of `self.verbose` checks. Is that OK for now?

File: perses/annihilation/ncmc_switching.py (hunk 5 above)
Comment (jchodera) [inline perses/annihilation/ncmc_switching.py]: Yep!  Will remove.

File: perses/annihilation/ncmc_switching.py (hunk 7 above)
Comment (pgrinaway) [inline perses/annihilation/ncmc_switching.py]: That's fine.

File: perses/samplers/samplers.py (hunk 4 above)
Comment (jchodera) [inline perses/samplers/samplers.py]: This implementation uses an `if` block, which should work fine to protect against `nan`s. 
The [openmmtools implementation of GHMC](https://github.com/choderalab/openmmtools/blob/master/openmmtools/integrators.py#L637-L639) does not have `nan` protection yet.
Do you want me to update that first and drop the code from here in this PR, or a separate PR?

File: perses/annihilation/ncmc_switching.py (hunk 6 above)
Comment (pgrinaway) [inline perses/annihilation/ncmc_switching.py]: Ok, if this is not ridiculous, it's probably preferable to do this inside an integrator?

File: devtools/conda-recipe/meta.yaml (hunk 2 above)
Comment (pgrinaway) [inline devtools/conda-recipe/meta.yaml]: Thanks!

File: perses/samplers/samplers.py (hunk 4 above)
Comment (pgrinaway) [inline perses/samplers/samplers.py]: Whichever you prefer--either is fine with me.

File: perses/annihilation/ncmc_switching.py (hunk 6 above)
Comment (jchodera) [inline perses/annihilation/ncmc_switching.py]: OK, I'll take a stab at that.

File: perses/samplers/samplers.py (hunk 4 above)
Comment (jchodera) [inline perses/samplers/samplers.py]: Let me break it out in a separate PR.

**Comment by jchodera** [comment]
This is ready to merge, @pgrinaway!

**Comment by jchodera** [comment]
Landscape is stalling on PRs for some reason, but all travis tests pass.

**Comment by pgrinaway** [comment]
Ok, LGTM, merging.
